File: java/program/Book.java
import java.util.ArrayList;

public class Book {
    // atribut-atribut private
    private String title;
    private ArrayList<Author> authorList;
    private Publisher publisher;
    private String ISBN;


    // constructor dengan parameter
    Book(String title, ArrayList<Author> authorList, Publisher publisher, String ISBN) {
        this.title = title;
        this.authorList = authorList;
        this.publisher = publisher;
        this.ISBN = ISBN;
    }

    // getter dan setter untuk title
    public void setTitle(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    // getter dan setter untuk author
    void setAuthor(ArrayList<Author> authorList) {
        this.authorList = authorList;
    }

    ArrayList<Author> getAuthor() {
        return authorList;
    }

    // getter dan setter untuk publisher
    void setPublisher(Publisher publisher) {
        this.publisher = publisher;
    }

    Publisher getPublisher() {
        return publisher;
    }

    // getter dan setter untuk ISBN
    public void setISBN(String ISBN) {
        this.ISBN = ISBN;
    }

    public String getISBN() {
        return ISBN;
    }
}

File: cpp/program/Book.cpp
#pragma once
#include <bits/stdc++.h>
#include "Author.cpp"
#include "Publisher.cpp"

using namespace std;

// kelas book
class Book
{
private:
    // atribut-atribut private
    string title;
    vector<Author> authorList;
    Publisher publisher;
    string ISBN;

public:
    // konstruktor kosong
    Book()
    {
    }

    // konstruktor dengan parameter untuk mengisi atribut
    Book(string title, vector<Author> authorList, Publisher publisher, string ISBN)
    {
        this->title = title;
        this->authorList = authorList;
        this->publisher = publisher;
        this->ISBN = ISBN;
    }

    // setter title
    void setTitle(string title)
    {
        this->title = title;
    }

    // getter title
    string getTitle()
    {
        return title;
    }

    // setter authorList
    void setAuthor(vector<Author> authorList)
    {
        this->authorList = authorList;
    }

    // getter authorList
    vector<Author> getAuthor()
    {
        return authorList;
    }

    // setter publisher
    void setPublisher(Publisher publisher)
    {
        this->publisher = publisher;
    }

    // getter publisher
    Publisher getPublisher()
    {
        return publisher;
    }

    // setter ISBN
    void setISBN(string ISBN)
    {
        this->ISBN = ISBN;
    }

    // getter ISBN
    string getISBN()
    {
        return ISBN;
    }

    ~Book()
    {
    }
};